### DISCUSSION TITLE
Section 8 lesson 186 downloader folder issues

### DISCUSSION THREAD

**Discussion by AxisMeetsWorld** [mrdbourke/pytorch-deep-learning#552] [Q&A] [answered]
Hello @mrdbourke, 
          I am hitting another roadblock here on the above lesson, and I can't see anything different about the way I have plugged the code in than you have. I ran all of the image path set up, ran the following lines that you have here:
`train_dir = image_path/ "train"
test_dir = image_path/ "test"`

Then, when testing you could see the paths:
**(PosixPath('data/pizza_steak_sushi/train'),
 PosixPath('data/pizza_steak_sushi/test'))**

Now, regardless of what I try to do, I get stuck on this line from the lesson

`# Create training and testing DataLoaders as well as get a list of class names
train_dataloader, test_dataloader, class_names = data_setup.create_dataloaders(train_dir=train_dir,
       ...........`
First, it said there was no drivepath (which seemed legit, and I am still puzzled by how **you didn't run into this error**)
"data/pizza_steak_sushi/train doesn't exist"

So, then I ran this code:
`if not train_dir.is_dir():
  train_dir.mkdir()`

`if not test_dir.is_dir():
  test_dir.mkdir()`   
just before the run. Well, now the error reads:

" Couldn't find any class folder in data/pizza_steak_sushi/train. "

So, I didn't see anything you may have run in [here](https://www.learnpytorch.io/06_pytorch_transfer_learning/) that showed any class establishment, so I am at a loss. What gives? I believe I am tapping into the same code you wrote when importing, so can you let me know what I could possibly have missed? Thanks as always!

**Comment by AxisMeetsWorld** [comment]
Just a quick update;
        When I tried to run the "Get some data" code, there was a line here that seemed to trip up:
`  with zipfile.ZipFile(data_path,"pizza/steak/sushi.zip", "r") as zip_ref:`
However, it doesn't seem to make much sense at all as the error stated:
" raise ValueError("ZipFile requires mode 'r', 'w', 'x', or 'a'")"
The code line seemed to specify that we were using "r", so why this error? It seems to be what is causing the error mentioned above, and when I did look into the class issue, I found that the files [needed to be unzipped](https://discuss.pytorch.org/t/filenotfounderror-couldnt-find-any-class-folder/138578/2), and due to what seems to be a nonsensical error above, it doesn't seem that the zip files are being unzipped. So, that seems to be the problem, but why on earth would I get that error when it quite obviously specified the use of **"r"** in the parameters for ZipFile?

**Reply by AxisMeetsWorld** [reply, marked as answer]
One more update:
      Ok, I just went and copied in what you had from the book version of the course and it worked. The only thing I can think of is that perhaps the spacing was off? Here is the copied in version from what you had :
`with zipfile.ZipFile(data_path / "pizza_steak_sushi.zip", "r") as zip_ref:`
When I ran the whole thing again, it worked. One thing I think I noticed was that in your file for the if....else conditions, you seemed to have 2 tabs, and I think I only had 1, under the `if:` and `else:` clauses. That seems like an odd error due to indents, but maybe you can help me understand what went on there? Thanks @mrdbourke
